docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -434,6 +468,21 @@ Creating a dataset from a generator function is also possible.
                                                     NULL, NULL);
     ```
 
+3. Setting `user_data` for the dataset can be done with `ml_train_dataset_set_property()`
+`dataset` does not take the ownership of `user_data`, so be sure to free after use.
+
+    ```c
+    /// assuming a user_data
```

Comment:
This is not clear 
```suggestion
    /// assuming user_data

Reply:
clarified to
`/// assuming user_data is retrived from `get_user_data()``